### PR TITLE
deduplicate and refine coordinate types

### PR DESCRIPTION
**Pull request by rotu**
Deduplicate and refine types `ICoordinateInterface`, `ICanonicalTileID`, `IMercatorCoordinate`, `ILngLat`

## Launch Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [X] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality.
 - [x] Document any changes to public APIs.
 - [x] Post benchmark scores.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

**Comment by HarelM** [comment]
THANKS! looks goods besides this small jsdocs comments.

File: src/coordinates/lng_lat.ts [hunk 1]
```diff
@@ -97,8 +98,8 @@ class LngLat {
      * Returns the approximate distance between a pair of coordinates in meters
      * Uses the Haversine Formula (from R.W. Sinnott, "Virtues of the Haversine", Sky and Telescope, vol. 68, no. 2, 1984, p. 159)
      *
-     * @param {LngLat} lngLat coordinates to compute the distance to
```
[the remark this answers is not in the document]
Comment (rotu) [reply src/coordinates/lng_lat.ts]: Thanks for pointing that out! I don't know how to build the docs, and it looks like it involves the https://github.com/maplibre/maplibre-gl-js-docs repository.

Maybe the workflow should be in a CONTRIBUTING.md file?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,7 @@
 ## main
+
+* Deduplicate and refine types `ICoordinateInterface`, `ICanonicalTileID`, `IMercatorCoordinate`, `ILngLat`
```
Comment (HarelM) [inline CHANGELOG.md:3]: I don't think this is interesting in term of changelog...

File: src/coordinates/lng_lat.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (HarelM) [reply src/coordinates/lng_lat.ts]: Sure, feel free to add that there.
Yes the docs are build after this is released, and maplibre-gl is released, it's a mess...

**Comment by HarelM** [comment]
@rotu is this still relevant?
Can we either close this PR or merge it?

**Comment by HarelM** [comment]
Replaced by #473

**Pull request closed without merging**